### PR TITLE
SNOW-870359: Integrate code coverage with CodeCov

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-jl**
Issue: https://github.com/snowflakedb/snowflake-sdks-drivers-issues-teamwork/issues/541

Added steps to upload code coverage report to Codecov in Github action and status badge in README.md
https://app.codecov.io/gh/snowflakedb/snowflake-connector-net/tree/SNOW-870359-codecov-integration

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -176,6 +186,22 @@ jobs:
         with:
           name: tests-performance
           path: Snowflake.Data.Tests/linux_${{ matrix.dotnet }}_${{ matrix.cloud_env }}_performance.csv
+      # codecov action sometimes fails but retry works - until action provides retry let's retry on our own https://github.com/codecov/codecov-action/issues/926
```
Comment (sfc-gh-dprzybysz) [inline .github/workflows/main.yml:189]: Retries may not work, but @sfc-gh-igarish can setup CodeCov token in secrets and we can then use the token as in https://github.com/snowflakedb/snowflake-connector-nodejs/pull/608

Comment (sfc-gh-igarish) [reply .github/workflows/main.yml:189]: Ok. setup the code cove token. Please check.

Comment (sfc-gh-ext-simba-jl) [reply .github/workflows/main.yml]: Codecov token is working. Thanks.

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Review by sfc-gh-mhofman — APPROVED** [review]

**Comment by sfc-gh-ext-simba-jl** [comment]
Not going to merge this PR yet since @sfc-gh-pfus has requested to remove the Snowflake.Data.Tests folder on Codecov.

**Comment by sfc-gh-pfus** [comment]
> Not going to merge this PR yet since @sfc-gh-pfus has requested to remove the Snowflake.Data.Tests folder on Codecov.

@sfc-gh-ext-simba-jl can you explain what you mean? I don't remember this topic, and .NET driver is not my area of experise...

**Comment by sfc-gh-ext-simba-jl** [comment]
Sorry tagged the wrong person. It should be @sfc-gh-pbulawa

**Comment by sfc-gh-dstempniak** [comment]
I see Snowflake.Data.Tests is excluded now:
https://app.codecov.io/gh/snowflakedb/snowflake-connector-net/tree/SNOW-870359-codecov-integration

@sfc-gh-ext-simba-jl please merge it

**Pull request merged**